src/core/Table/actionHandlers/expandHandler.ts

```diff
@@ -12,13 +12,13 @@ export const onExpandHandler = <T extends Record<string, unknown>>(
     tableState?: TableState<T>,
   ) => void,
 ) => {
-  if (!instance?.rows.length) {
+  if (!instance?.page.length) {
     onExpand?.([], newState);
     return;
   }
 
   const expandedData: T[] = [];
-  instance.rows.forEach((row) => {
+  instance.page.forEach((row) => {
```

Comment:
I will take a look at this change later.

Reply:
Changed to `preFilteredFlatRows` because when expanded `rows` list doesn't have expanded/visible sub-rows.